cpp/src/fil/infer.cu

```diff
@@ -395,29 +398,29 @@ struct tree_aggregator_t<NITEMS, GROVE_PER_CLASS_MANY_CLASSES> {
   __device__ __forceinline__ tree_aggregator_t(predict_params params,
                                                void* accumulate_workspace,
                                                void* finalize_workspace)
-      : per_class_margin((vec<NITEMS, float>*)accumulate_workspace) {
-      tmp_storage(finalize_workspace),
-      num_classes(num_classes_) {
+    : per_class_margin((vec<NITEMS, float>*)accumulate_workspace),
+      tmp_storage(params.predict_proba ? per_class_margin + num_classes
+                                       : finalize_workspace),
+      num_classes(params.num_classes) {
     for (int c = threadIdx.x; c < num_classes; c += blockDim.x)
-      per_class_value[c] = vec<NITEMS, float>(0);
+      per_class_margin[c] = vec<NITEMS, float>(0);
     // __syncthreads() is called in infer_k
   }
 
   __device__ __forceinline__ void accumulate(
-    vec<NITEMS, float> single_tree_prediction, int tree) {
+    vec<NITEMS, float> single_tree_prediction, int tree, int num_rows) {
+    for (int i = num_rows; i < NITEMS; ++i) acc[i] = 0.0f;
```

Comment:
fixing a bug which somehow did not trigger the test in the previous PR. It triggers it now.